web/src/settings_users.js

```diff
@@ -633,3 +643,44 @@ export function set_up_bots() {
         settings_bots.add_a_new_bot();
     });
 }
+
+export async function fetch_presence_user_setting() {
+    if (page_params.is_spectator) {
+        return;
+    }
+
+    await new Promise((resolve) => {
+        channel.post({
+            url: "/json/users/me/presence",
+            data: {
+                status: compute_active_status(),
+                ping_only: false,
+                last_update_id: -1,
+                history_limit_days: 3650,
```

Comment:
Rather than framing this as a "10 years of history", I think you should just pass an even greater value, like 9999 to fully express the intent that we just want all history (not some arbitrary cutoff).

Reply:
sure!